NativeScript/NativeScript.mm

```diff
@@ -21,18 +21,68 @@ @implementation Config
 
 @implementation NativeScript
 
-std::unique_ptr<Runtime> runtime_;
+extern char defaultStartOfMetadataSection __asm("section$start$__DATA$__TNSMetadata");
 
-- (instancetype)initWithConfig:(Config*)config {
+- (void)runScriptString: (NSString*) script runLoop: (BOOL) runLoop {
+
+    std::string cppString = std::string([script UTF8String]);
+    runtime_->RunScript(cppString);
     
+    if (runLoop) {
+        CFRunLoopRunInMode(kCFRunLoopDefaultMode, 0, true);
+    }
+
+
+    tns::Tasks::Drain();
+
+}
+
+std::unique_ptr<Runtime> runtime_;
+
+- (void)runMainApplication {
+    runtime_->RunMainScript();
+
+    CFRunLoopRunInMode(kCFRunLoopDefaultMode, 0, true);
+
+    tns::Tasks::Drain();
+}
+
+- (bool)liveSync {
+    if (runtime_ == nullptr) {
+        return false;
+    }
+
+    Isolate* isolate = runtime_->GetIsolate();
+    return tns::LiveSync(isolate);
+}
+
+- (void)shutdownRuntime {
+    if (runtime_ != nullptr) {
+        Isolate* isolate = runtime_->GetIsolate();
+        {
+            v8::Locker l(isolate);
+            v8::Isolate::Scope isolate_scope(isolate);
+            v8::HandleScope handle_scope(isolate);
+            
+            isolate->Dispose();
```

Comment:
this likely shouldn't be done here (the runtime destructor is responsible for deleting this isolate)